Animal table v2 PR

mertkantaroglu opened the pull request:
Hello,

In this table I added species subject and new animals to the table. Afterward I looked for the queries and got the required data from the table per below screenshots.

Looking forward to your feedback. 

Thanks,

![Screen Shot 2023-04-11 at 19 22 39](https://user-images.githubusercontent.com/44959066/231233286-62838094-ccb6-4ef2-93fc-12d178aaebf6.png)
![Screen Shot 2023-04-11 at 19 22 53](https://user-images.githubusercontent.com/44959066/231233297-a9d7dbfc-689e-482e-a5ac-55b4cbe829f3.png)
![Screen Shot 2023-04-11 at 19 27 00](https://user-images.githubusercontent.com/44959066/231233300-1c755d6f-b9e0-46f2-9211-191ce6a3eab1.png)
![Screen Shot 2023-04-11 at 19 28 48](https://user-images.githubusercontent.com/44959066/231233303-d0812c53-fb0a-4001-8141-9174f36b6689.png)
![Screen Shot 2023-04-11 at 19 31 36](https://user-images.githubusercontent.com/44959066/231233305-eba90e22-026d-49c6-b8b2-ac666ca3a9db.png)
![Screen Shot 2023-04-11 at 19 44 46](https://user-images.githubusercontent.com/44959066/231233307-81e10a0a-862e-4441-9f67-9de5a7db6020.png)
![Screen Shot 2023-04-11 at 15 27 16](https://user-images.githubusercontent.com/44959066/231233333-189061e2-6752-4a1d-8e40-b5200e1160b7.png)

mertkantaroglu committed:
Create SQL files and update Readme file

mertkantaroglu committed:
Fix PR issue

mertkantaroglu committed:
Fix Readme error

mertkantaroglu committed:
Update animal table and queries

bobb-Rob commented on schema.sql:
```diff
@@ -0,0 +1,9 @@
+  escape_attempts INTEGER,
+  neutered BOOLEAN,
+  weight_kg DECIMAL(5,2),
+  species VARCHAR(50)
```
- [x] At this point of the project, the animal table had already been created, what was expected from you is to alter the table by adding a column using the `ALTER TABLE animal ADD species varchar(255);` 

Kindly add the column using the ALTER command instead of adding it like you did in line 8.

mahbubul14 replied:
> * [x]  At this point of the project, the animal table had already been created, what was expected from you is to alter the table by adding a column using the `ALTER TABLE animal ADD species varchar(255);`
> 
> Kindly add the column using the ALTER command instead of adding it as you did in line 8.

- [x] You have implemented nearly all of the suggestions made by the previous reviewer. However, it appears that you may have overlooked one recommendation, which was to create a new column named `species` with a data type of string, as requested, using the `alter` command. :+1:

bobb-Rob commented on queries.sql:
```diff
@@ -0,0 +1,49 @@
+SELECT * FROM animals WHERE name != 'Gabumon';
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
```
- [x] As mentioned in the project requirement instructions, you are expected to write a query to verify your UPDATE before rollback and after rollback. Like so 👇🏾 

```sql
BEGIN; -- start transaction

 UPDATE animals
 SET species = 'unspecified'; -- make change

SELECT species from animals; -- verify that change was made
ROLLBACK;

SELECT species from animals; -- verify that change was undone
``` 
You did it well in your query screenshot but not here in the code file.
Kindly do add that to this file.

bobb-Rob commented on queries.sql:
```diff
@@ -0,0 +1,49 @@
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+ROLLBACK;
```
- [x] This code from line 14 to 16 should be inside a separate transaction and there should be a query to verify that the change was made **_before_** and **_after_** the commit. If you take a look at your [second screenshot](https://user-images.githubusercontent.com/44959066/231163127-68ff8a1a-e99e-4021-b16b-aeb3fb65cf91.png) - there is a warning after the commit that reads - **_`there no transaction in progress`_** and this simply means that the command above is not done within a transaction as required. 
- [x] The ROLLBACK command on line 17 is not necessary. please remove it.

Kindly fix this

bobb-Rob commented on queries.sql:
```diff
@@ -0,0 +1,49 @@
+COMMIT;
+ROLLBACK;
+
+BEGIN TRANSACTION;
+DELETE FROM animals;
+ROLLBACK;
```
- [x] Similarly, please write a query into this file to verify that the data was deleted before the rollback and restored after the rollback as you did in the screenshot.

bobb-Rob commented on queries.sql:
```diff
@@ -0,0 +1,49 @@
+BEGIN TRANSACTION;
+DELETE FROM animals;
+ROLLBACK;
+
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+
+BEGIN TRANSACTION;
+SAVEPOINT my_sp1;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO SAVEPOINT my_sp1;
+
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
```
- [x] You did very well on this code, but there is a little issue to note. The `BEGIN;` on line 25, should have come before the code on line 23 because **_deleting all animals born after Jan 1st, 2022_** was suppose to be inside a transaction as well.

```sql
    BEGIN;

    DELETE FROM animals WHERE  date_of_birth > '2022/01/01' ;
    SAVEPOINT SP1;

    UPDATE animals SET weight_kg = weight_kg * -1; 
    ROLLBACK TO SP1;
    UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0; 

    COMMIT;
```
The [fourth ](https://user-images.githubusercontent.com/44959066/231163118-47ad7096-d0bb-4a9b-a2b5-54cf15c053d2.png) and [fifth](https://user-images.githubusercontent.com/44959066/231163115-23b41f45-4b5d-44f3-b889-f28f7a877f3f.png) screenshots should be one screenshot since both belong to the same transaction. Kindly fix this and remember to update your screenshots

bobb-Rob reviewed (changes requested):
## STATUS: CHANGES REQUIRED ♻️
Hi @mertkantaroglu, 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>
![](https://camo.githubusercontent.com/34482800e1b92cb79ddca06edb4051edc407cb190f896e1e949e596aa9073851/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f556461535559436c723932636a57706556782f67697068792e676966)

## Highlights
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

mertkantaroglu committed:
Fix PR issues

mertkantaroglu commented:
Thanks for your detailed review and suggestions @bobb-Rob 👍 💯

mahbubul14 reviewed (changes requested):
Hi @mertkantaroglu,

Great work on making the changes requested by the previous reviewer 👏

You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

# Required Changes ♻️

- [ ] Kindly check the inline comments under the review. :arrow_down: 

###  To Highlights
:heavy_check_mark: No linter error.
:heavy_check_mark: Proper Github flow used.
:heavy_check_mark: Descriptive PR title & description.

### Optional suggestions

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

* N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also reach me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

mertkantaroglu commented:
Hello @mahbubul14,

I already implemented the item you mentioned at my first screenshot. You can check it in below picture. At the second table you can see that I put the query, I altered the table and added the 'species' with varchar(255).

![Screen Shot 2023-04-11 at 19 22 39](https://user-images.githubusercontent.com/44959066/231247263-c0a2d415-2413-457e-93c1-ad1aa5afe023.png)

mertkantaroglu committed:
Fix PR issues

AshakaE commented:
Hi @mertkantaroglu , 

### Highlights ✨ 

- Thanks for making the requested changes 👏 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Proceed to the next level 👏 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

mahbubul14 commented:
> Hello @mahbubul14,
> 
> I already implemented the item you mentioned at my first screenshot. You can check it in below picture. At the second table you can see that I put the query, I altered the table and added the 'species' with varchar(255).
> 
> ![Screen Shot 2023-04-11 at 19 22 39](https://user-images.githubusercontent.com/44959066/231247263-c0a2d415-2413-457e-93c1-ad1aa5afe023.png)

@mertkantaroglu , It appears that you have added it to your `local machine/ terminal`, but have not yet included it in the schema.sql file.
